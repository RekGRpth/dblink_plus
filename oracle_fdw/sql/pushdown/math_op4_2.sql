\c contrib_regression_utf8

SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';

-- '/' operator
-- 153 : int2div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_smallint_1 / number_smallint_1 = 1;
-- 154 : int4div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_integer_1 / number_integer_1 = 1;
-- 172 : int24div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_smallint_1 / number_integer_1 = 1;
-- 173 : int42div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_integer_1 / number_smallint_1 = 1;
-- 203 : float4div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_float_real_1 / binary_float_real_1 = 1;
-- 217 : float8div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_double_precision_1 / binary_double_precision_1 = 1;
-- 280 : float48div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_float_real_1 / binary_double_precision_1 = 1;
-- 284 : float84div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_double_precision_1 / binary_float_real_1 = 1;
-- 466 : int8div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_bigint_1 / number_bigint_1 = 1;
-- 840 : int82div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_bigint_1 / number_smallint_1 = 1;
-- 948 : int28div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_smallint_1 / number_bigint_1 = 1;
-- 1277 : int84div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_bigint_1 / number_integer_1 = 1;
-- 1281 : int48div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_integer_1 / number_bigint_1 = 1;
-- 1727 : numeric_div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 <> 0 AND number_numeric3_1 / number_numeric3_1 = 1;

-- 153 : int2div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_smallint_1 / number_smallint_1 = 1;
-- 154 : int4div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_integer_1 / number_integer_1 = 1;
-- 172 : int24div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_smallint_1 / number_integer_1 = 1;
-- 173 : int42div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_integer_1 / number_smallint_1 = 1;
-- 203 : float4div
SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_float_real_1 / binary_float_real_1 = 1;
-- 217 : float8div
SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_double_precision_1 / binary_double_precision_1 = 1;
-- 280 : float48div
SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_float_real_1 / binary_double_precision_1 = 1;
-- 284 : float84div
SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_double_precision_1 / binary_float_real_1 = 1;
-- 466 : int8div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_bigint_1 / number_bigint_1 = 1;
-- 840 : int82div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_bigint_1 / number_smallint_1 = 1;
-- 948 : int28div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_smallint_1 / number_bigint_1 = 1;
-- 1277 : int84div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_bigint_1 / number_integer_1 = 1;
-- 1281 : int48div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_integer_1 / number_bigint_1 = 1;
-- 1727 : numeric_div
SELECT id FROM all_type WHERE number_numeric3_1 <> 0 AND number_numeric3_1 / number_numeric3_1 = 1;

