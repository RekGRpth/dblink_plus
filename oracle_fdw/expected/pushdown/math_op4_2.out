\c contrib_regression_utf8
SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';
-- '/' operator
-- 153 : int2div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_smallint_1 / number_smallint_1 = 1;
                                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                                           
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" <> 0) AND (("number_smallint_1" / "number_smallint_1") = 1))
(2 rows)

-- 154 : int4div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_integer_1 / number_integer_1 = 1;
                                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                                         
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" <> 0) AND (("number_integer_1" / "number_integer_1") = 1))
(2 rows)

-- 172 : int24div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_smallint_1 / number_integer_1 = 1;
                                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                                          
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" <> 0) AND (("number_smallint_1" / "number_integer_1") = 1))
(2 rows)

-- 173 : int42div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_integer_1 / number_smallint_1 = 1;
                                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                                          
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" <> 0) AND (("number_integer_1" / "number_smallint_1") = 1))
(2 rows)

-- 203 : float4div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_float_real_1 / binary_float_real_1 = 1;
                                                                                                                                                                                                                                                             QUERY PLAN                                                                                                                                                                                                                                                              
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" <> 0) AND (("binary_float_real_1" / "binary_float_real_1") = 1))
(2 rows)

-- 217 : float8div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_double_precision_1 / binary_double_precision_1 = 1;
                                                                                                                                                                                                                                                                      QUERY PLAN                                                                                                                                                                                                                                                                       
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" <> 0) AND (("binary_double_precision_1" / "binary_double_precision_1") = 1))
(2 rows)

-- 280 : float48div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_float_real_1 / binary_double_precision_1 = 1;
                                                                                                                                                                                                                                                                   QUERY PLAN                                                                                                                                                                                                                                                                    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" <> 0) AND (("binary_float_real_1" / "binary_double_precision_1") = 1))
(2 rows)

-- 284 : float84div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_double_precision_1 / binary_float_real_1 = 1;
                                                                                                                                                                                                                                                                QUERY PLAN                                                                                                                                                                                                                                                                 
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" <> 0) AND (("binary_double_precision_1" / "binary_float_real_1") = 1))
(2 rows)

-- 466 : int8div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_bigint_1 / number_bigint_1 = 1;
                                                                                                                                                                                                                                                       QUERY PLAN                                                                                                                                                                                                                                                        
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" <> 0) AND (("number_bigint_1" / "number_bigint_1") = 1))
(2 rows)

-- 840 : int82div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_bigint_1 / number_smallint_1 = 1;
                                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                                          
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" <> 0) AND (("number_bigint_1" / "number_smallint_1") = 1))
(2 rows)

-- 948 : int28div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_smallint_1 / number_bigint_1 = 1;
                                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                                         
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" <> 0) AND (("number_smallint_1" / "number_bigint_1") = 1))
(2 rows)

-- 1277 : int84div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_bigint_1 / number_integer_1 = 1;
                                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                                         
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" <> 0) AND (("number_bigint_1" / "number_integer_1") = 1))
(2 rows)

-- 1281 : int48div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_integer_1 / number_bigint_1 = 1;
                                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                                        
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" <> 0) AND (("number_integer_1" / "number_bigint_1") = 1))
(2 rows)

-- 1727 : numeric_div
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 <> 0 AND number_numeric3_1 / number_numeric3_1 = 1;
                                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                                           
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_numeric3_1" <> 0) AND (("number_numeric3_1" / "number_numeric3_1") = 1))
(2 rows)

-- 153 : int2div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_smallint_1 / number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 154 : int4div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_integer_1 / number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 172 : int24div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_smallint_1 / number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 173 : int42div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_integer_1 / number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 203 : float4div
SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_float_real_1 / binary_float_real_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 217 : float8div
SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_double_precision_1 / binary_double_precision_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 280 : float48div
SELECT id FROM all_type WHERE binary_double_precision_1 <> 0 AND binary_float_real_1 / binary_double_precision_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 284 : float84div
SELECT id FROM all_type WHERE binary_float_real_1 <> 0 AND binary_double_precision_1 / binary_float_real_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 466 : int8div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_bigint_1 / number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 840 : int82div
SELECT id FROM all_type WHERE number_smallint_1 <> 0 AND number_bigint_1 / number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 948 : int28div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_smallint_1 / number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1277 : int84div
SELECT id FROM all_type WHERE number_integer_1 <> 0 AND number_bigint_1 / number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1281 : int48div
SELECT id FROM all_type WHERE number_bigint_1 <> 0 AND number_integer_1 / number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1727 : numeric_div
SELECT id FROM all_type WHERE number_numeric3_1 <> 0 AND number_numeric3_1 / number_numeric3_1 = 1;
 id 
----
  5
  7
(2 rows)

